source/grid/grid_tools_dof_handlers.cc

```diff
@@ -2063,8 +2063,8 @@ namespace GridTools
     Assert(0 <= direction && direction < space_dim,
            ExcIndexRange(direction, 0, space_dim));
 
-    Assert(pairs1.size() == pairs2.size(),
-           ExcMessage("Unmatched faces on periodic boundaries"));
+    // Assert(pairs1.size() == pairs2.size(),
+    //        ExcMessage("Unmatched faces on periodic boundaries"));
```

Comment:
I have commented this and the next assert, because both cases might happen for periodic distributed coarse grids. 

**Note**:  [step-5](https://github.com/peterrum/dealii-pft/blob/master/step-5/src/main.cpp) deals with periodic boundaries. I am not very happy with the fact that two times `add_periodicity()` has to be called. I might be able to make the periodicity part of `ConstructionData`...